File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/models/PlantsWaste.java
package io.github.thunkier.thunkmetrc.wrapper.models;
import com.fasterxml.jackson.annotation.JsonProperty;public class PlantsWaste {
    @JsonProperty("GrowthPhase")
    public Integer growthPhase;
    @JsonProperty("Label")
    public String label;
    @JsonProperty("LocationId")
    public Integer locationId;
    @JsonProperty("LocationName")
    public String locationName;
    @JsonProperty("PlantId")
    public Integer plantId;
    @JsonProperty("PlantWasteId")
    public Integer plantWasteId;
    @JsonProperty("StateName")
    public String stateName;
    @JsonProperty("SublocationId")
    public Integer sublocationId;
    @JsonProperty("SublocationName")
    public String sublocationName;
    @JsonProperty("WasteAmount")
    public Integer wasteAmount;
    @JsonProperty("WasteUnitOfMeasureAbbreviation")
    public String wasteUnitOfMeasureAbbreviation;
}

File: sdks/thunkmetrc-rs/wrapper/src/models/plants_waste.rs
use serde::{Deserialize, Serialize};
#[allow(unused_imports)]
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PlantsWaste {
    #[serde(rename = "GrowthPhase", skip_serializing_if = "Option::is_none")]
    pub growth_phase: Option<i64>,
    #[serde(rename = "Label", skip_serializing_if = "Option::is_none")]
    pub label: Option<String>,
    #[serde(rename = "LocationId", skip_serializing_if = "Option::is_none")]
    pub location_id: Option<i64>,
    #[serde(rename = "LocationName", skip_serializing_if = "Option::is_none")]
    pub location_name: Option<String>,
    #[serde(rename = "PlantId", skip_serializing_if = "Option::is_none")]
    pub plant_id: Option<i64>,
    #[serde(rename = "PlantWasteId", skip_serializing_if = "Option::is_none")]
    pub plant_waste_id: Option<i64>,
    #[serde(rename = "StateName", skip_serializing_if = "Option::is_none")]
    pub state_name: Option<String>,
    #[serde(rename = "SublocationId", skip_serializing_if = "Option::is_none")]
    pub sublocation_id: Option<i64>,
    #[serde(rename = "SublocationName", skip_serializing_if = "Option::is_none")]
    pub sublocation_name: Option<String>,
    #[serde(rename = "WasteAmount", skip_serializing_if = "Option::is_none")]
    pub waste_amount: Option<i64>,
    #[serde(rename = "WasteUnitOfMeasureAbbreviation", skip_serializing_if = "Option::is_none")]
    pub waste_unit_of_measure_abbreviation: Option<String>,
}
